tests/js/server/aql/aql-graph-traverser.js

```diff
@@ -2943,6 +2943,53 @@ function brokenGraphSuite() {
   };
 }
 
+/*
+ *       A
+ *     ↙ ↲ ↘  // Edge from A to the edge connecting "A->B"
+ *   B       C
+ */
+function edgeConnectedFromVertexToEdge() {
+  return {
+    setUpAll: function () {
+      cleanup();
+
+      vc = db._createDocumentCollection(vn, { numberOfShards: 4 });
+      ec = db._createEdgeCollection(en, { numberOfShards: 4 });
+
+      vertex.A = vc.save({ _key: 'A' })._id;
+      vertex.B = vc.save({ _key: 'B' })._id;
+      vertex.C = vc.save({ _key: 'C' })._id;
+
+      // collections and directions
+
+      edge.AB = ec.save(vertex.A, vertex.B, {});
+      edge.AC = ec.save(vertex.A, vertex.C, {});
+      edge.AAB = ec.save(vertex.A, edge.AB, {}); // Edge from A to the edge connecting "A->B"
+    },
+
+    tearDownAll: function () {
+      cleanup();
+    },
+
+
+    testConnectedEdgeToAnotherEdge: function () {
+      var queries = [
+        [`WITH ${vn}, ${en} FOR x,y,z IN 1..10 OUTBOUND @start @@ec return x`, false]
```

Comment:
Can you please also add a test on a named graph.
It is okay to just reuse the data, as this is a query planning issue.

Reply:
Added two more tests.